application/src/main/java/org/togetherjava/tjbot/commands/filesharing/FileSharingMessageListener.java

```diff
@@ -0,0 +1,183 @@
+        if (author.isBot() || event.isWebhookMessage()) {
+            return;
+        }
+
+        if (!isHelpThread(event)) {
+            return;
+        }
+
+        List<Message.Attachment> attachments =
+                event.getMessage().getAttachments().stream().filter(attachment -> {
+                    String extension = attachment.getFileExtension();
+                    if (extension == null) {
+                        return false;
+                    }
+                    return extensionFilter.contains(extension);
+                }).toList();
+
+        processAttachments(event, attachments);
+    }
+
+    private void processAttachments(@NotNull MessageReceivedEvent event,
+            @NotNull List<Message.Attachment> attachments) {
+        Map<String, GistFile> filesAsJson = new HashMap<>();
+
+        for (Message.Attachment attachment : attachments) {
+            attachment.retrieveInputStream()
+                .thenApply(this::readAttachment)
+                .thenAccept(content -> filesAsJson.put(createFileName(attachment),
+                        new GistFile(content)))
+                .join();
+        }
+
+        GistFiles files = new GistFiles(filesAsJson);
+        GistRequest request = new GistRequest(event.getAuthor().getName(), false, files);
+        String url = uploadToGist(request);
+        sendResponse(event, url);
+    }
+
+    private @NotNull String readAttachment(@NotNull InputStream stream) {
+        try {
+            return new String(stream.readAllBytes(), StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private @NotNull String createFileName(@NotNull Message.Attachment attachment) {
+        String fileName = attachment.getFileName();
+        String fileExtension = attachment.getFileExtension();
+
+        if (fileExtension == null || fileExtension.equals("txt")) {
+            fileExtension = "java";
+        } else if (fileExtension.equals("fxml")) {
+            fileExtension = "xml";
+        }
+
+        fileName = fileName.substring(0, fileName.lastIndexOf(".")) + "." + fileExtension;
+        return fileName;
+    }
+
+    private @NotNull String uploadToGist(@NotNull GistRequest jsonRequest) {
+        String body;
+        try {
+            body = JSON.writeValueAsString(jsonRequest);
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException("Couldn't parse json request!", e);
+        }
+
+        HttpRequest request = HttpRequest.newBuilder()
+            .uri(URI.create(SHARE_API))
+            .header("Accept", "application/json")
+            .header("Authorization", "token " + gistApiKey)
+            .POST(HttpRequest.BodyPublishers.ofString(body))
+            .build();
+
+        HttpResponse<String> apiResponse;
+        try {
+            apiResponse = CLIENT.send(request, HttpResponse.BodyHandlers.ofString());
```

Comment:
the blocking send now also happens on the JDA thread. so u definitely have to ensure that the whole flow (`processAttachments`) now happens on another thread (see previous comment)